python_modules/dagster/dagster/_core/definitions/assets/job/asset_job.py

```diff
@@ -238,7 +238,18 @@ def get_asset_graph_for_job(
         create_unexecutable_external_asset_from_assets_def,
     )
 
-    selected_keys = selection.resolve(parent_asset_graph)
+    try:
+        selected_keys = selection.resolve(parent_asset_graph)
+    except KeyError as e:
+        raise DagsterInvalidDefinitionError(
+            "Selected keys must be a subset of existing executable asset keys."
+            f" Invalid selected keys: {e.args}",
+        )
+    except Exception:
+        raise DagsterInvalidDefinitionError(
+            f"Invalid asset selection: {selection}",
+        )
```

Comment:
I think we want to use `raise from` so the underlying exception with details is still available https://docs.python.org/3/tutorial/errors.html#exception-chaining